### ISSUE TITLE
Run MS tests strategy to cover all the test cases

### ISSUE THREAD

**Issue by yitzhak12** [red-hat-storage/ocs-ci#6881]
Some tier4 tests(tier4a, tier4b, and maybe tier4c) are applicable for provider and consumer clusters, like `test_nodes_restart` or `test_rolling_shutdown_and_recovery_in_controlled_fashion`. 
That means if we specify that the consumer cluster is first, those tests will run only on the consumer cluster and not on the provider. 

Moreover, if we switch to the provider cluster in some tests and the consumer cluster in other tests, in that case, we must also check that the Sanity helper function is configured correctly on the current running cluster.
For example, we have a run with one consumer and one provider. And the default cluster is the consumer. Let’s take the test `test_osd_node_restart_and_check_osd_pods_status` for this example.
In the setup function, it inits the Sanity helper for the consumer. When we switch to the provider cluster in the test, it checks the provider functionality, which can lead to validating the wrong cluster details.

In summary, if we want to cover all cases in tier4 we have two options:

1. Run one time when the consumer cluster is first(as we currently do) and another when the provider cluster is first.
2. Switch to the relevant cluster(consumer or provider) in every test by demand and init the sanity helper in the function. We can use something like [here](https://drive.google.com/file/d/1xYkJ0UriK1KGvO4zrKaKmOjDSxh7t23Y/view?usp=share_link)

I already discussed it with @ebenahar, and he prefers option 2(with a small addition).

**Comment by yitzhak12** [comment]
Please Let me know WDYT

**Comment by ebenahar** [comment]
+1 on option 2

**Comment by nehaberry** [comment]
@jilju can you also share your input

**Comment by jilju** [comment]
I'd prefer option 2.

**Comment by yitzhak12** [comment]
Yes. @ebenahar and I also thought about it. But we still need to init the sanity function in the test itself and not in the `setup` method, right?

**Comment by fbalak** [comment]
> Yes. @ebenahar and I also thought about it. But we still need to init the sanity function in the test itself and not in the `setup` method, right?

It's true that it would be problematic. Also the context would need to be used correctly in fixtures.

**Comment by dahorak** [comment]
I don't have strong opinion here, because it will depends on how many such tests we have and how many tests we have which are tight to only one role (provider/consumer) and how the numbers are spread across the various testsuites etc.

But I would also incline to the option 2 with the enhancements as mentioned above, with having in mind some limitation and specifics, mainly that it will means, that the selected `TEST_EXECUTION_STRATEGY` in the multicluster job will not ensure, that all the tests will be executed on the selected role (because the tests will switch to different cluster based on the test itself or the marks for the tests, not based on which cluster is default).
On the other hand I think it will have quite a nice benefit if we will use @pytest.mark.parametrize, that in the results we should directly see if the particular test were executed on provider or consumer cluster.

**Comment by yitzhak12** [comment]
Okay, thanks for the input @dahorak
